docs/design/initial-design.md

```diff
@@ -0,0 +1,66 @@
+# Armada - Jupyter Deployment Design
+
+## Problem Description
+
+Armada's ability to deploy across clusters should be exploited to allow for the deployment of per-user JupyterLab instances.
+
+With minimal configuration changes to the Armada chart, a user should be able to deploy a JupyterLab instance that is accessible from their browser using with a URL that is unique to them. This should all be done from one of the Armada Clients.
+
+## Proposed Solution
+
+Users will be able to write a configuration file for setting up the jupyterlab instance, and then deploy this with a single command. A single command will also allow for cancelling the deployment.
+
+## Current Questions
+
+- Should we write this in Python, or Go? (Go would have a single executable, which is probably better for the end user)
+- How will we handle the authentication between the armada client and the armada server?
+- How will we allow for shutting down the jupyterlab instance? We will need some way of remembering / storing armada job ID's or applying a custom label to the job.
```

Comment:
Cancelling a job-set will bring down jupyter.  

I would probably suggest a timeout as we don't want people to have infinite time on these machines.

Reply:
I'm not sure about a timeout. Some of these AI models can take a significant time to run, so we don't want users getting cut off midway through training. Thoughts?

Reply:
Yea it isn't great.  But I also don't think we want infinite time on these jobs.  In supercompuing centers, they will usually put a wall time limit on jobs like this so people don't have a permanent shell taking up compute resources.  

At the NIH, they have a configurable walltime limit of 36 hours for jobs like this.

Typically you would only use jobs like this for debugging and I would suggest longer running training jobs probably don't need interactively.

Reply:
That's a good point. I will add a timeout option to the config.